main.cpp

```diff
@@ -0,0 +1,284 @@
+/*
+Aplicaci�n para armar torneos de f�tbol. 
+Integrantes:
+
+Karen Cano
+Jos� Cullen
+Melisa Cruz Mart�nez
+Soledad Ferrari 
+*/
+
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+#include <iomanip>
+using namespace std;
+
+char NOMBRE_ARCHIVO [20] = "EquiposDeFutbol.bin";
+const int NOT_FOUND = -1;
+const int MAX_EQUIPOS = 100;
+FILE *archivo;
+
+
+struct Equipo{
+	char id [4];
+	char nombre[32];
+	int potenciaAtaque;
+	int potenciaDefensa;
+};
+
+void MostrarMenu(bool permiteSalir);
+void AgregarEquipo();
+void ListarEquipos();
+void EjecutarOpcion(int opcion);
+int PedirOpcion(int opciones, bool permiteSalir);
+int CantidadEquipos();
+int PosicionDelEquipoEnArchivo(char id[4]);
+void EliminarEquipo();
+void SetearListaDeEquipos(Equipo equipos []);
+void ReescribirArchivo(Equipo equipos[], int tamanio);
+void AgregarEquipoAlArchivo(Equipo equipo);
+Equipo CrearEquipo(bool permiteNuevo);
+
+
+int main(int argc, char** argv) {
+	int numero;
+	
+	archivo = fopen(NOMBRE_ARCHIVO,"ab");
+	fclose(archivo);
+	
+	MostrarMenu(true);	
+	return 0;
+}
+
+
+/*
+	Muestra el men� principal
+	permiteSalir	booleano que indica si debe mostrar dicha opci�n.
+*/
+void MostrarMenu(bool permiteSalir){
+	cout<<"Bienvenido al Equipos de Futbol"<<endl<<endl;
+	cout<<"1- Agregar Equipo."<<endl;
+	cout<<"2- Eliminar Equipo."<<endl;
+	cout<<"3- Ver Equipos."<<endl;
+
+	if(permiteSalir){
+		cout<<"0- Salir."<<endl;	
+	}	
+	EjecutarOpcion(PedirOpcion(3, permiteSalir));
+}
+
+/* 
+	Pide una opci�n, validando que sea correcta. 
+	return opci�n elegida
+*/
+int PedirOpcion(int opciones, bool permiteSalir){
+	int opcion = NOT_FOUND;
+	while(opcion < 0){
+		cout<<"Ingrese la opci�n deseada: ";
+		cin>>opcion;
+		if( (opcion == 0 && !permiteSalir) || opcion < 0 || opcion > opciones ){
+			cout<<"La opci�n ingresada no es v�lida"<<endl;
+			opcion = NOT_FOUND;
+		}
+	}
+	return opcion;
+}
+
+/*
+	Ejecuta la opci�n elegida. 
+*/
+void EjecutarOpcion(int opcion){
+	switch(opcion){
+		case 1:
+			AgregarEquipo();
+			MostrarMenu(false);			
+			break;
+		case 2:
+			EliminarEquipo();
+			MostrarMenu(false);			
+				break;
+		case 3:
+			ListarEquipos();
+			MostrarMenu(false);
```

Comment:
Sii.. Estoy de acuerdo, pero eso decía el enunciado.. Supongo que porque más tarde hay algo para que permita salir, por eso dejé el código preparado para eso
José María Cullen
________________________________
De: Emanuel<mailto:notifications@github.com>
Enviado el: ‎28/‎06/‎2015 17:23
Para: josecullen/algoritmos-utn<mailto:algoritmos-utn@noreply.github.com>
Asunto: Re: [algoritmos-utn] Correcciones (#1)

> +/*
> +     Ejecuta la opci�n elegida.
> +*/
> +void EjecutarOpcion(int opcion){
> +     switch(opcion){
> +             case 1:
> +                     AgregarEquipo();
> +                     MostrarMenu(false);
> +                     break;
> +             case 2:
> +                     EliminarEquipo();
> +                     MostrarMenu(false);
> +                             break;
> +             case 3:
> +                     ListarEquipos();
> +                     MostrarMenu(false);

``permiteSalir`` siempre va en false, así que no se puede salir nunca de la aplicación.

---
Reply to this email directly or view it on GitHub:
https://github.com/josecullen/algoritmos-utn/pull/1/files#r33426533